src/net_multi.c

```diff
@@ -53,15 +55,74 @@ ksnMultiClass *ksnMultiInit(ksnMultiData *md, void *user_data) {
             // Start network
             ksnetEvMgrRun(ke);
             
-            // Start event manager 
-            if(md->run && i == md->num - 1) ev_run(ke->ev_loop, 0);          
+            // Start event manager
+            if(md->run && i == md->num - 1) ev_run(ke->ev_loop, 0);
         }
         
     }
     
     return km;
 }
 
+/**
+ * Add new network
+ * 
+ * @param km
+ * @param event_cb Event callback
+ * @param host Host name
+ * @param port Port number
+ * @param network Network name
+ */ 
+void teoMultiAddNet(ksnMultiClass *km, multi_event_cb e_cb, const char *host, int port, const char *network) {
+    ksnetEvMgrClass *ke_last = teoMultiGetByNumber(km, km->num-1);
+
+    // We need to update count of networks for old networks
+    ksnMultiUpdateCountNetworks(km, km->num + 1);
+
+    // If port is 0 use port+2 from last network
+    if(!port) {
+        port = ke_last->ksn_cfg.port+2;
+    }
+
+    ksnetEvMgrClass *ke_new = ksnetEvMgrInitPort(ke_last->argc, ke_last->argv,
+                e_cb, READ_OPTIONS|READ_CONFIGURATION, port, NULL);
+    
+    // Set network parameters
+    ke_new->km = km; // Pointer to multi net module
+    ke_new->n_num = km->num - 1; // Set network number
```

Comment:
I don't get it, why is there `km->num - 1` and not `+1`?
Where it was incremented?

Reply:
> I don't get it, why is there `km->num - 1` and not `+1`?
> Where it was incremented?

in function ksnMultiUpdateCountNetworks(km, km->num + 1);